File: api/src/main/java/com/api/app/entity/MemberBase.java
package com.api.app.entity;

import io.swagger.v3.oas.annotations.media.Schema;
import lombok.Getter;
import lombok.Setter;
import org.apache.ibatis.type.Alias;

/**
 * @author system
 * @version 1.0
 * @since 2025-10-27
 */
@Alias("MemberBase")
@Getter
@Setter
public class MemberBase extends SystemEntity {
    private static final long serialVersionUID = 9012345678901234567L;

    @Schema(description = "회원번호")
    private String memberNo;

    @Schema(description = "회원명")
    private String memberName;

    @Schema(description = "전화번호")
    private String phone;

    @Schema(description = "이메일")
    private String email;

    @Schema(description = "비밀번호")
    private String password;

    @Schema(description = "회원상태코드")
    private String memberStatusCode;
}

File: api/src/main/kotlin/com/api/app/entity/MemberBase.kt
package com.api.app.entity

import jakarta.persistence.Column
import jakarta.persistence.Entity
import jakarta.persistence.Id
import jakarta.persistence.Table

@Entity
@Table(name = "MEMBER_BASE")
class MemberBase : SystemEntity() {

    @Id
    @Column(name = "MEMBER_NO", length = 15, nullable = false)
    var memberNo: String = ""

    @Column(name = "MEMBER_NAME", nullable = false, length = 30)
    var memberName: String = ""

    @Column(name = "PHONE", length = 20)
    var phone: String? = null

    @Column(name = "EMAIL", nullable = false, length = 100)
    var email: String = ""

    @Column(name = "PASSWORD", nullable = false, length = 200)
    var password: String = ""

    @Column(name = "MEMBER_STATUS_CODE", nullable = false, length = 3)
    var memberStatusCode: String = ""
}
